### PR TITLE
Generator: Add Custom Extension Findable

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description

From the spec findable `x-go-findable` `1` and `2` it's building the custom findable we want.
By default, it's remain `name 1`  and `id 2` but now you can for example give

for elastic-ip you can point:
`1 - id`
`2 - ip`

and findable will generate a findable func with `idOrIP` parameter to be found 

## Checklist
(For exoscale contributors)

* [x] Changelog updated (under *Unreleased* block)
* [x] For a new resource or new attributes: test added/updated

### CODE REVIEW

**Review by elkezza — APPROVED** [review]
lgtm :)

File: v3/README.md [hunk 1]
```diff
@@ -70,6 +70,23 @@ func main() {
...
+### Findable
+
+Most of the list request `ListX()` return a type containing the list of the resource requested and a method `FindX()` to be able to retrieve a resource by its `name` or `id` most of the time.
+
+```Golang
+instances, err := client.ListInstancePools(ctx)
+if err != nil {
+	log.Fatal(err)
+}
+pool, err := instances.FindInstancePool("my-pool-example")
```
Comment (kobajagi) [inline v3/README.md:82]:
```suggestion
pools, err := client.ListInstancePools(ctx)
if err != nil {
	log.Fatal(err)
}
pool, err := pools.FindInstancePool("my-pool-example")
```

File: v3/README.md [hunk 2]
```diff
@@ -70,6 +70,23 @@ func main() {
 }	
 ```
 
+### Findable
+
+Most of the list request `ListX()` return a type containing the list of the resource requested and a method `FindX()` to be able to retrieve a resource by its `name` or `id` most of the time.
```
Comment (kobajagi) [inline v3/README.md:75]: `X` in `FindX()` is not really needed. All functions could just be named `Find()` as they are method of specific struct.
It's a breaking change though, you can decide if it's worth changing or leave it as is.

Comment (pierre-emmanuelJ) [reply v3/README.md:75]: For the moment we will leave as it is to not breaking change....

File: v3/README.md [hunk 3]
```diff
@@ -87,3 +104,53 @@ GENERATOR_DEBUG=client make generate > test/client.go
...
+	Generated code
+	```Golang
+	type Endpoint string
+
+	type Zone struct {
+		APIEndpoint Endpoint // Here is the generated type definition.
+		...
+	}
+	```
+- `x-go-findable` to specify which fields in the findable resource to fetch
```
Comment (kobajagi) [inline v3/README.md:129]: From my understanding of the code, there are some limitations worth documenting:
- `x-go-findable` can only be set on string types;
- max 2 attributes can have a `x-go-findable`;
- values of `x-go-findable` can be either `"1"` or `"2"` , where former value determines string will be search there first.

**Review by kobajagi — APPROVED** [review]

**Pull request merged**